### PR TITLE
fix: propagate condition to sqs queue policy for sqssubscription

### PR DESCRIPTION
**Pull request by hawflau**
*Issue #, if available:*

*Description of changes:*

This fixes a bug of not having `Condition` propagated to the generated `AWS::SQS::QueuePolicy` resource if a SNS event source is used with `SqsSubscription` in a `AWS::Serverless::Function` resource.

Also updated the black commands in `Makefile` to check only `.py` files

*Description of how you validated changes:*

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [ ] Update documentation
- [x] Verify transformed template deploys and application functions as expected

*Examples?*

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,10 +9,10 @@ test:
 	pytest --cov samtranslator --cov-report term-missing --cov-fail-under 95 tests
 
 black:
-	black setup.py samtranslator/* tests/* bin/*.py
+	black setup.py samtranslator/**/*.py tests/**/*.py bin/*.py
```
Comment (hoffa) [inline Makefile:12]: Why is this change necessary? This also assumes the shell supports `**` (which isn't in POSIX `sh`), which e.g. in Bash is only enabled if `globstar` is set. AFAIK `make` won't necessarily default to Bash.

Tangentially, can we instead simplify this to `black setup.py samtranslator tests bin` and change include/exclude behavior in the config?

Comment (hawflau) [reply Makefile:12]: I intended to skip checking the .pyc files. But looking at the doc, yeah you are right, it's not necessary.
Will revert this commit

**Review by hoffa — APPROVED** [review]

**Review by wchengru — APPROVED** [review]

File: tests/translator/input/function_event_conditions.yaml [hunk 2]
```diff
@@ -79,6 +79,22 @@ Resources:
...
+        SNSTopicWithSQSSubscription:
+          Type: SNS
+          Properties:
+            Topic:
+              Ref: Notifications
+            SqsSubscription:
+              QueueArn: !GetAtt Queue.Arn
+              QueueUrl: !Ref Queue
+
+        AnotherSNSWithSQSSubscription:
```
Comment (elbayaaa) [inline tests/translator/input/function_event_conditions.yaml:91]: I'd prefer a more descriptive name to immediately figure out what is the difference between this one and the one above, what each of them tests.

**Review by elbayaaa — APPROVED** [review]
